### PR TITLE
README: updated text

### PR DESCRIPTION
**Pull request by emmanuelsearch**
a somewhat less "experimental" readme

### CODE REVIEW

**Comment by jkarinkl** [comment]
See also this [template](https://github.com/cncf/project-template/blob/main/README-template.md) for README's. Would be nice to work to such an extended README in time.

**Comment by emmanuelsearch** [comment]
OK, nice pointer indeed, thanks! Although : for now we would be missing some pieces eg a new contributor guide.

**Comment by jkarinkl** [comment]
Indeed, see issue #151.

File: README.md [hunk 1]
```diff
@@ -5,38 +5,43 @@
...
+ - more to come soon.
 
 ## Status
 
-The current iteration of RIOT-rs combines [embassy](https://embassy.dev/) with
-a preemptive scheduler and adds some integration and build system work.
+**This is currently work-in-progress. Expect missing functionalities and frequent changes!** 
+If you are not so adventurous, but nevertheless looking for a way 
+to run your Rust module on a microcontroller, you could try to 
+glue it directly on top of the [embassy HAL](https://github.com/embassy-rs/embassy),
```
Comment (kaspar030) [inline README.md:39]:
```suggestion
glue it directly on top of [embassy](https://github.com/embassy-rs/embassy), 
```

(embassy is more than "a HAL")

File: README.md [hunk 2]
```diff
@@ -5,38 +5,43 @@
...
+In particular, RIOT-rs aims for:
 
-## Goals
-
-- improve RIOT using the merits of Rust.
-- provide a "rusty" development workflow (e.g., using cargo / crates.io)
-- provide a nice Rust API, framework and collection of crates suitable for
-  embedded development
-- rewrite parts of RIOT in Rust to improve robustness and maintainability
+- **code portability** across all supported hardware, via consistent memory/energy efficient APIs;
```
[the remark this answers is not in the document]
Comment (emmanuelsearch) [reply README.md:23]: sounds good.

File: README.md (hunk 1 above)
Comment (emmanuelsearch) [reply README.md:39]: Yup

File: README.md [hunk 3]
```diff
@@ -5,38 +5,43 @@
...
 
-## Goals
-
-- improve RIOT using the merits of Rust.
-- provide a "rusty" development workflow (e.g., using cargo / crates.io)
-- provide a nice Rust API, framework and collection of crates suitable for
-  embedded development
-- rewrite parts of RIOT in Rust to improve robustness and maintainability
+- **application code portability** across all supported hardware, via consistent memory/energy efficient APIs;
+- programming based either/both on **async & preemptive scheduling** paradigms, the latter using formally verified modules using [hax](https://hacspec.org/blog/posts/hax-v0-1/);
```
Comment (kaspar030) [inline README.md:24]: would it make sense to split this in two, async first (mentioning that it is based on embassy) and second "preemptive scheduling", (using verified modules based on hax)?

**Comment by kaspar030** [comment]
It think this is an improvement, we can merge as is. I'm somewhat missing the added value over embassy (that RIOT-rs is heavily based on embassy, but provides abstraction & integration on top).

File: README.md [hunk 4]
```diff
@@ -5,38 +5,43 @@
...
+operating system adequate for cybersecure, memory-safe, low-power IoT.
 
-**This is highly experimental. Expect heavy changes and breakage!**
+RIOT-rs is based on Rust from the ground up, and uses formal verification for critical modules. 
+For a more verbose rant, see this
+[manifesto](https://future-proof-iot.github.io/RIOT-rs/dev/manifesto.html).
 
-If you're looking for a more production ready way of writing RIOT applications
-in Rust, check out [riot-wrappers](https://gitlab.com/etonomy/riot-wrappers).
+Overall, RIOT-rs aims for a 'batteries-included' experience, on par
```
Comment (jkarinkl) [inline README.md:15]: "RIOT-rs aims for" may be rewritten in firmer terms. Maybe "RIOT-rs will give you"? or "gives you".

**Review by jkarinkl — COMMENTED** [review]
Some suggestions for textual change. Overall: nice text. 
It is good to keep in mind who the intended audience for this text is. What is their level of knowledge and what kind of wording is appropriate? Should the sentences be shorter and with easier wording? I find this difficult to assess, as I do not yet know the target group well. This might be a point of consideration in reviewing the text.

File: README.md (hunk 4 above)
Comment (emmanuelsearch) [reply README.md]: thanks for the suggestion, now incorporated in the text.

File: README.md [hunk 5]
```diff
@@ -5,38 +5,43 @@
 
 > Rust & RIOT combined for ergonomic embedded development
 
-This is an experimental project to provide a nice base OS for embedded
-development on low-end IoT devices (with some kilobytes of RAM/flash, think
-Cortex-M). It combines the awesome Rust embedded ecosystem with
-[RIOT](https://github.com/RIOT-OS/RIOT).
+RIOT-rs is a project aiming to provide a general-purpose
+operating system adequate for cybersecure, memory-safe, low-power IoT.
```
[the remark this answers is not in the document]
Comment (emmanuelsearch) [reply README.md]: thanks for the suggestion, now incorporated in the text.

File: README.md [hunk 6]
```diff
@@ -5,38 +5,43 @@
...
 
-**This is highly experimental. Expect heavy changes and breakage!**
+RIOT-rs is based on Rust from the ground up, and uses formal verification for critical modules.
```
[the remark this answers is not in the document]
Comment (emmanuelsearch) [reply README.md]: I think this is the level of technicality we aim for, for the moment. The target audience should know what we mean by that.

File: README.md (hunk 3 above)
Comment (emmanuelsearch) [reply README.md]: good suggestion. Now split.

**Comment by emmanuelsearch** [comment]
> It think this is an improvement, we can merge as is. I'm somewhat missing the added value over embassy (that RIOT-rs is heavily based on embassy, but provides abstraction & integration on top).

OK. I made an attempt at that in the fixed text.

**Comment by emmanuelsearch** [comment]
@kaspar030 see lines 18-21

File: README.md [hunk 7]
```diff
@@ -5,38 +5,49 @@
...
-in Rust, check out [riot-wrappers](https://gitlab.com/etonomy/riot-wrappers).
-
-## Goals
-
-- improve RIOT using the merits of Rust.
-- provide a "rusty" development workflow (e.g., using cargo / crates.io)
-- provide a nice Rust API, framework and collection of crates suitable for
-  embedded development
-- rewrite parts of RIOT in Rust to improve robustness and maintainability
+RIOT-rs is an operating system for cybersecure, memory-safe, low-power IoT.
```
[the remark this answers is not in the document]
Comment (emmanuelsearch) [reply README.md:8]: We could. To enphasize the fact that we are networked, we could then expand the IoT acronym?

File: README.md [hunk 8]
```diff
@@ -5,38 +5,49 @@
...
+RIOT-rs is based on Rust from the ground up, and uses formal verification
+for critical modules. For a more verbose rant, see this
```
[the remark this answers is not in the document]
Comment (emmanuelsearch) [reply README.md:10]: well rants are not always negative ;) I'm not attached to the term. What alternative wording do your propose? How about: "for a more verbose explanation, see this..."

[the remark this answers is not in the document]
Comment (bergzand) [reply README.md:10]: I would go for a generic "For more information, see…"

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (emmanuelsearch) [reply README.md]: Addressed

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (emmanuelsearch) [reply README.md]: Addressed!

**Comment by kaspar030** [comment]
Let's take this.

**Comment by kaspar030** [comment]
Next time, use fixup commits, then squash just before merging (which can be done by the final reviewer). That way we keep the history of changes.

**Comment by kaspar030** [comment]
And, thanks! :smile:

**Pull request merged**